pkg/controller/verticalpodautoscaler/verticalpodautoscaler_controller.go

```diff
@@ -757,10 +758,56 @@ func (r *Reconciler) VPAPodSpec(vpa *autoscalingv1.VerticalPodAutoscalerControll
 	return spec
 }
 
+// RecommenderControllerPodSpec returns the expected podSpec for the Recommender Controller deployment belonging
+// to the given VerticalPodAutoscalerController.
+func (r *Reconciler) RecommenderControllerPodSpec(vpa *autoscalingv1.VerticalPodAutoscalerController, params ControllerParams) *corev1.PodSpec {
+	spec := r.VPAPodSpec(vpa, params)
+
+	// Allow the user to override the resources of the container
+	if (!reflect.DeepEqual(vpa.Spec.ContainerOverrides.Recommender.Resources, corev1.ResourceRequirements{})) {
+		spec.Containers[0].Resources = vpa.Spec.ContainerOverrides.Recommender.Resources
+	}
+
+	// Append user args to our container args
+	if len(vpa.Spec.ContainerOverrides.Recommender.Args) > 0 {
+		spec.Containers[0].Args = append(spec.Containers[0].Args, vpa.Spec.ContainerOverrides.Recommender.Args...)
```

Comment:
I was appending the user args, but I could probably do something like keda does here if we want it to work like that https://github.com/openshift/custom-metrics-autoscaler-operator/blob/ea0c16951aabdeaecba690d718d2304bc779bc5d/controllers/keda/transform/transform.go#L820